src/utils/upload/putPartInS3.js

```diff
@@ -0,0 +1,44 @@
+import axios from 'axios';
+import fetchAPI from 'utils/http/fetchAPI';
+
+const MAX_RETRIES = 3;
+
+const putPartInS3 = async (
+  blob, uploadParams, partNumber, abortController, onUploadProgress, currentRetry = 0,
+) => {
+  try {
+    const signedUrl = await getSignedUrlForPart(uploadParams, partNumber);
+
+    return await axios.request({
+      method: 'put',
+      data: blob,
+      url: signedUrl,
+      signal: abortController.signal,
+      headers: {
+        'Content-Type': 'application/octet-stream',
+      },
+      onUploadProgress,
+    });
+  } catch (e) {
+    if (currentRetry < MAX_RETRIES) {
+      return await putPartInS3(
+        blob, uploadParams, partNumber, abortController, onUploadProgress, currentRetry + 1,
+      );
+    }
+
+    throw e;
+  }
+};
+
+const getSignedUrlForPart = async (uploadParams, partNumber) => {
+  const {
+    experimentId, uploadId, bucket, key,
+  } = uploadParams;
+
+  const queryParams = new URLSearchParams({ bucket, key });
+  const url = `/v2/experiments/${experimentId}/upload/${uploadId}/part/${partNumber}/signedUrl?${queryParams}`;
```

Comment:
Can this be generic for any file? Seems like its specific for experiments only.

Reply:
Without the `experimentId` we can't control that the client has authorization to access that experiment they are modifying